infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/LogReplicationAckReader.java

```diff
@@ -327,19 +335,61 @@ public void setBaseSnapshot(long baseSnapshotTimestamp) {
         this.baseSnapshotTimestamp = baseSnapshotTimestamp;
     }
 
+    public void markSnapshotSyncInfoCompleted() {
```

Comment:
why do we need a special method to mark the sync as complete?  Cant it be inferred from the remainingReplicationEntries?

Reply:
remainingReplicationEntries is not 100% accurate and will have 0-15s latency to update status. So I am using the FSM transitions to update status.